src/functions.js

```diff
@@ -37,13 +37,26 @@ function get_click_handler( item ) {
 			return;
 		}
 
+		// Get the current provider.
+		let provider = this.controller.state().get( 'library' ).props.get( 'provider' );
+		if ( ! provider && Object.keys( AMF_DATA.providers ).length > 0 ) {
+			provider = Object.keys( AMF_DATA.providers )[0];
+		}
+
```

Comment:
I was wondering about that, whether to handle it in the JS or not. If the provider is undefined/null when we post to the back end then it will fail at that point and return an error response. That felt like the most straight forward way to use a single error handler for this function.

We shouldn't default to local as that can be switched off. Really this case should never happen where there's no provider unless someone intentionally prevents local media _and_ does not register any custom providers.

Reply:
Yes, it's an edge case, but may happen. We could ensure that the `providers` is never empty, but would fall back to an array with just the blank provider in it? 🤔

Reply:
Ah it's not an array right now it's an object of providers / associative array. I think it's probably better to avoid the blank provider approach.

Could do something like `provider = Object.keys( AMF_DATA.providers )[0] || null;` instead?